### ISSUE TITLE
wrong histograms

### ISSUE THREAD

**Issue by greaber** [lanpa/tensorboard-pytorch#42]
Hi, I am having problems plotting histograms.  I think there is a very good chance that it is not because of a bug in tensorboard-pytorch, but I'm not sure what I could be doing wrong, and I'm not sure where to ask, so if someone could help I would appreciate it.

I am trying to plot histograms of the gradients like this:

```
loss.backward()
for n, p in filter(lambda np: np[1].grad is not None, spectral_model.named_parameters()):
    print(n, p.grad.data.min(), p.grad.data.max())
    summary_writer.add_histogram(n, p.grad.data.cpu().numpy(), global_step=step)
```
The mins and maxes show that the values are all between -.15 and .15 (and in fact most values are much closer to zero than that).  But the histograms seem to show that all the values are located at one extremely high value, like 3.01e+18:

![image](https://user-images.githubusercontent.com/8763901/31584773-71fc0df4-b1bd-11e7-9fde-8ad45eafee71.png)

**Comment by lanpa** [comment]
http://tensorboard-pytorch.readthedocs.io/en/latest/tensorboard.html#tensorboardX.SummaryWriter.add_histogram

Hi, will additional parameter `bins=range(...)` help?

**Comment by greaber** [comment]
I didn't try this exactly, but I tried some of the string values for bins listed in the numpy docs.  Setting bins='auto' causes the program to crash with MemoryError coming from a call to np.zeros (even though the machine has plenty of memory).  Setting bins='sturges' fixes the problem and leads to nice histograms.  But it is pretty strange; I can't think of anything really unusual about my usage that would make the default and 'auto' histograms fail for me if they aren't broken for everyone else.

**Comment by greaber** [comment]
Actually 'sturges' doesn't do that great of a job.  It uses too few bins and doesn't zoom in enough.  But at least it is not completely wrong like the default histograms.

**Comment by greaber** [comment]
I also tried 'doane' and 'fd'.  doane seems about the same as sturges, and fd crashes with a memory error like auto.

**Comment by lanpa** [comment]
Using `auto` will explode the memory (should be numpy bug). see #1 . From you last graph, I think `bins=np.arange(-0.003, 0.003, 0.0001)` might do the work.

**Comment by greaber** [comment]
Thanks.  For now I guess I will just stick to 'doane' since I don't want to hand set the binning for each parameter.  Actually, I am finding the distributions tab more useful than the histograms tab for my puproses anyway (basically I am interested in monitoring spiking gradients that require clipping and maybe also gradients going to zero).  We can close this issue if you want.  There is definitely some buggy behavior with the default binning, but if tensorboard is doing the binning by default I guess it is probably a tensorboard issue.
